From 2f411e4d1e199152e371728023d2d6d214f2c1df Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Wed, 7 Aug 2019 21:35:17 +0900
Subject: [PATCH] disable ENABLE_IPTS_DEBUG

---
 drivers/misc/ipts/ipts.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/ipts/ipts.h b/drivers/misc/ipts/ipts.h
index 9c34b55ff..583823289 100644
--- a/drivers/misc/ipts/ipts.h
+++ b/drivers/misc/ipts/ipts.h
@@ -26,7 +26,7 @@
 #include "ipts-state.h"
 #include "ipts-binary-spec.h"
 
-#define ENABLE_IPTS_DEBUG		/* enable IPTS debug */
+// #define ENABLE_IPTS_DEBUG		/* enable IPTS debug */
 
 #ifdef ENABLE_IPTS_DEBUG
 
-- 
2.22.0

