From f5d7c4ceebf6744a09412ef8b9228f0c621bb583 Mon Sep 17 00:00:00 2001
From: Jake Day <jake@ninebysix.com>
Date: Thu, 27 Jun 2019 11:59:48 -0400
Subject: [PATCH 04/11] cameras

---
 drivers/staging/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/Makefile b/drivers/staging/Makefile
index 0da0d3f0b..b01e8a139 100644
--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -49,3 +49,4 @@ obj-$(CONFIG_XIL_AXIS_FIFO)	+= axis-fifo/
 obj-$(CONFIG_EROFS_FS)		+= erofs/
 obj-$(CONFIG_FIELDBUS_DEV)     += fieldbus/
 obj-$(CONFIG_KPC2000)		+= kpc2000/
+obj-$(CONFIG_VIDEO_OV5693)     += ov5693/
-- 
2.22.0

