From ab8d1b218da2ee582a8a7069dfe7dd8553b929e0 Mon Sep 17 00:00:00 2001
From: "Hayataka@kitakar5525" <34676735+kitakar5525@users.noreply.github.com>
Date: Wed, 13 Feb 2019 23:34:14 +0900
Subject: [PATCH] ipts: uncomment downstream_hpd_needs_d0

---
 drivers/gpu/drm/i915/intel_dp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 7cec187da..2b0a5ccbe 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2798,8 +2798,8 @@ void intel_dp_sink_dpms(struct intel_dp *intel_dp, int mode)
 		return;
 
 	if (mode != DRM_MODE_DPMS_ON) {
-		//if (downstream_hpd_needs_d0(intel_dp))
-		//	return;
+		if (downstream_hpd_needs_d0(intel_dp))
+			return;
 
 		ret = drm_dp_dpcd_writeb(&intel_dp->aux, DP_SET_POWER,
 					 DP_SET_POWER_D3);
-- 
2.20.1

