From f37467a27ddbb22102dc7e850769d4d5a5f1cf00 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Fri, 2 Aug 2019 05:16:16 +0900
Subject: [PATCH] ipts: print debug of lib/idr.c for 5.3rc

---
 lib/idr.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/idr.c b/lib/idr.c
index 66a374892..96ae21602 100644
--- a/lib/idr.c
+++ b/lib/idr.c
@@ -393,6 +393,8 @@ int ida_alloc_range(struct ida *ida, unsigned int min, unsigned int max,
 	unsigned long flags;
 	struct ida_bitmap *bitmap, *alloc = NULL;
 
+	pr_alert("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	if ((int)min < 0)
 		return -ENOSPC;
 
@@ -405,6 +407,9 @@ int ida_alloc_range(struct ida *ida, unsigned int min, unsigned int max,
 	bitmap = xas_find_marked(&xas, max / IDA_BITMAP_BITS, XA_FREE_MARK);
 	if (xas.xa_index > min / IDA_BITMAP_BITS)
 		bit = 0;
+
+	pr_alert("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	if (xas.xa_index * IDA_BITMAP_BITS + bit > max)
 		goto nospc;
 
@@ -413,6 +418,7 @@ int ida_alloc_range(struct ida *ida, unsigned int min, unsigned int max,
 
 		if (bit < BITS_PER_XA_VALUE) {
 			bit = find_next_zero_bit(&tmp, BITS_PER_XA_VALUE, bit);
+			pr_alert("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
 			if (xas.xa_index * IDA_BITMAP_BITS + bit > max)
 				goto nospc;
 			if (bit < BITS_PER_XA_VALUE) {
@@ -436,6 +442,7 @@ int ida_alloc_range(struct ida *ida, unsigned int min, unsigned int max,
 
 	if (bitmap) {
 		bit = find_next_zero_bit(bitmap->bitmap, IDA_BITMAP_BITS, bit);
+		pr_alert("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
 		if (xas.xa_index * IDA_BITMAP_BITS + bit > max)
 			goto nospc;
 		if (bit == IDA_BITMAP_BITS)
-- 
2.22.0

