From a92b4a2c0191ad419215ad964e1f9c2ad7b101da Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Fri, 2 Aug 2019 06:42:02 +0900
Subject: [PATCH 05/12] Resolved build time errors for ipts-5.2

---
 drivers/gpu/drm/i915/intel_ipts.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ipts.c b/drivers/gpu/drm/i915/intel_ipts.c
index 3d3c35398..9fb92c760 100644
--- a/drivers/gpu/drm/i915/intel_ipts.c
+++ b/drivers/gpu/drm/i915/intel_ipts.c
@@ -27,6 +27,9 @@
 #include <linux/intel_ipts_if.h>
 #include <drm/drmP.h>
 
+#include "gt/intel_context.h"
+#include "gem/i915_gem_context.h"
+
 #include "intel_guc_submission.h"
 #include "i915_drv.h"
 
@@ -86,7 +89,7 @@ static intel_ipts_object_t *ipts_object_create(size_t size, u32 flags)
 	}
 
 	/* Allocate the new object */
-	gem_obj = i915_gem_object_create(dev_priv, size);
+	gem_obj = i915_gem_object_create_shmem(dev_priv, size);
 	if (gem_obj == NULL) {
 		ret = -ENOMEM;
 		goto err_out;
@@ -136,8 +139,8 @@ static int ipts_object_pin(intel_ipts_object_t* obj,
 	struct drm_i915_private *dev_priv = to_i915(intel_ipts.dev);
 	int ret = 0;
 
-	if (ipts_ctx->ppgtt) {
-		vm = &ipts_ctx->ppgtt->vm;
+	if (ipts_ctx->vm) {
+		vm = ipts_ctx->vm;
 	} else {
 		vm = &dev_priv->ggtt.vm;
 	}
@@ -192,7 +195,7 @@ static int create_ipts_context(void)
 		goto err_unlock;
 	}
 
-	ce = intel_context_pin(ipts_ctx, dev_priv->engine[RCS0]);
+	ce = intel_context_create(ipts_ctx, dev_priv->engine[RCS0]);
 	if (IS_ERR(ce)) {
 		DRM_ERROR("Failed to create intel context (error %ld)\n",
 			  PTR_ERR(ce));
@@ -241,7 +244,7 @@ static void destroy_ipts_context(void)
 
 	ipts_ctx = intel_ipts.ipts_context;
 
-	ce = intel_context_lookup(ipts_ctx, dev_priv->engine[RCS0]);
+	ce = intel_context_create(ipts_ctx, dev_priv->engine[RCS0]);
 
 	/* Initialize the context right away.*/
 	ret = i915_mutex_lock_interruptible(intel_ipts.dev);
@@ -452,8 +455,8 @@ static int intel_ipts_map_buffer(u64 gfx_handle, intel_ipts_mapbuffer_t *mapbuf)
 		obj->cpu_addr = ipts_object_map(obj);
 	}
 
-	if (ipts_ctx->ppgtt) {
-		vm = &ipts_ctx->ppgtt->vm;
+	if (ipts_ctx->vm) {
+		vm = ipts_ctx->vm;
 	} else {
 		vm = &dev_priv->ggtt.vm;
 	}
-- 
2.22.0

