From b847881beb086a07d6feabdea71a98cfc04dc6f2 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Fri, 2 Aug 2019 07:13:13 +0900
Subject: [PATCH] ipts: try to deal with runtime errors

---
 drivers/gpu/drm/i915/intel_ipts.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_ipts.c b/drivers/gpu/drm/i915/intel_ipts.c
index 9fb92c760..f8175195f 100644
--- a/drivers/gpu/drm/i915/intel_ipts.c
+++ b/drivers/gpu/drm/i915/intel_ipts.c
@@ -203,6 +203,16 @@ static int create_ipts_context(void)
 		goto err_unlock;
 	}
 
+	ret = __intel_context_do_pin(ce);
+	if (ret) {
+		DRM_ERROR("context pinning failed : %d\n", ret);
+		return ret;
+	}
+
+	/* intel_context_is_pinned */
+	if (atomic_read(&ce->pin_count))
+		pr_alert("DEBUG: context is not pinned!\n");
+
 	ret = execlists_context_deferred_alloc(ce, ce->engine);
 	if (ret) {
 		DRM_DEBUG("lr context allocation failed : %d\n", ret);
-- 
2.22.0

