From e39e5e5ca0bfc3a8bcdd3784a4c94bda8643de88 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Thu, 11 Jul 2019 08:50:43 +0900
Subject: [PATCH] nvme: add pr_info debug

---
 drivers/nvme/host/pci.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index ebbf6aba8..879ce14b5 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -2843,6 +2843,8 @@ static int nvme_deep_state(struct nvme_dev *dev)
 	struct nvme_ctrl *ctrl = &dev->ctrl;
 	int ret = -EBUSY;;
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_start_freeze(ctrl);
 	nvme_wait_freeze(ctrl);
 	nvme_sync_queues(ctrl);
@@ -2886,6 +2888,8 @@ static int nvme_make_operational(struct nvme_dev *dev)
 {
 	struct nvme_ctrl *ctrl = &dev->ctrl;
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	if (nvme_set_features(ctrl, NVME_FEAT_POWER_MGMT, dev->last_ps,
 			      NULL, 0, NULL) == 0)
 		return 0;
@@ -2898,6 +2902,8 @@ static int nvme_simple_resume(struct device *dev)
 	struct pci_dev *pdev = to_pci_dev(dev);
 	struct nvme_dev *ndev = pci_get_drvdata(pdev);
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_reset_ctrl(&ndev->ctrl);
 	return 0;
 }
@@ -2907,6 +2913,8 @@ static int nvme_resume(struct device *dev)
 	struct pci_dev *pdev = to_pci_dev(dev);
 	struct nvme_dev *ndev = pci_get_drvdata(pdev);
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	return pm_resume_via_firmware() || !ndev->ctrl.npss ?
 		nvme_simple_resume(dev) : nvme_make_operational(ndev);
 }
-- 
2.22.0

