From e4a204d3b2affe1a4f27a55c338e499891cceb35 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Fri, 6 Sep 2019 12:32:20 +0900
Subject: [PATCH] nvme: add print debug for suspend

---
 drivers/nvme/host/pci.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index 732d5b63e..69fcaa61b 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -2847,6 +2847,8 @@ static int nvme_resume(struct device *dev)
 	struct nvme_dev *ndev = pci_get_drvdata(to_pci_dev(dev));
 	struct nvme_ctrl *ctrl = &ndev->ctrl;
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	if (ndev->last_ps == U32_MAX ||
 	    nvme_set_power_state(ctrl, ndev->last_ps) != 0)
 		nvme_reset_ctrl(ctrl);
@@ -2878,9 +2880,11 @@ static int nvme_suspend(struct device *dev)
 	if (pm_suspend_via_firmware() || !ctrl->npss ||
 	    !pcie_aspm_enabled(pdev) ||
 	    (ndev->ctrl.quirks & NVME_QUIRK_SIMPLE_SUSPEND)) {
+		pr_info("DEBUG: nvme_suspend(): using generic PCI power settings\n");
 		nvme_dev_disable(ndev, true);
 		return 0;
 	}
+	pr_info("DEBUG: nvme_suspend(): using host managed power settings\n");
 
 	nvme_start_freeze(ctrl);
 	nvme_wait_freeze(ctrl);
@@ -2923,6 +2927,8 @@ static int nvme_simple_suspend(struct device *dev)
 {
 	struct nvme_dev *ndev = pci_get_drvdata(to_pci_dev(dev));
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_dev_disable(ndev, true);
 	return 0;
 }
@@ -2932,6 +2938,8 @@ static int nvme_simple_resume(struct device *dev)
 	struct pci_dev *pdev = to_pci_dev(dev);
 	struct nvme_dev *ndev = pci_get_drvdata(pdev);
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_reset_ctrl(&ndev->ctrl);
 	return 0;
 }
-- 
2.23.0

