From 5e269d48f4536b9ce94bb7e865c39b45bfcbe1f8 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Wed, 31 Jul 2019 08:55:22 +0900
Subject: [PATCH] nvme: add print debug for suspend

---
 drivers/nvme/host/pci.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index aea3b42a5..779d3d063 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -2860,6 +2860,8 @@ static int nvme_resume(struct device *dev)
 	struct nvme_dev *ndev = pci_get_drvdata(to_pci_dev(dev));
 	struct nvme_ctrl *ctrl = &ndev->ctrl;
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	if (pm_resume_via_firmware() || !ctrl->npss ||
 	    nvme_set_power_state(ctrl, ndev->last_ps) != 0)
 		nvme_reset_ctrl(ctrl);
@@ -2882,9 +2884,13 @@ static int nvme_suspend(struct device *dev)
 	 * device fully.
 	 */
 	if (pm_suspend_via_firmware() || !ctrl->npss) {
+		pr_info("DEBUG: Passed %s %d, (pm_suspend_via_firmware() || !ctrl->npss)==true \n",__FUNCTION__,__LINE__);
+		pr_info("DEBUG: nvme_suspend(): using generic PCI power settings\n");
 		nvme_dev_disable(ndev, true);
 		return 0;
 	}
+	pr_info("DEBUG: Passed %s %d, (pm_suspend_via_firmware() || !ctrl->npss)==false \n",__FUNCTION__,__LINE__);
+	pr_info("DEBUG: nvme_suspend(): using host managed power settings\n");
 
 	nvme_start_freeze(ctrl);
 	nvme_wait_freeze(ctrl);
@@ -2928,6 +2934,8 @@ static int nvme_simple_suspend(struct device *dev)
 {
 	struct nvme_dev *ndev = pci_get_drvdata(to_pci_dev(dev));
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_dev_disable(ndev, true);
 	return 0;
 }
@@ -2937,6 +2945,8 @@ static int nvme_simple_resume(struct device *dev)
 	struct pci_dev *pdev = to_pci_dev(dev);
 	struct nvme_dev *ndev = pci_get_drvdata(pdev);
 
+	pr_info("DEBUG: Passed %s %d \n",__FUNCTION__,__LINE__);
+
 	nvme_reset_ctrl(&ndev->ctrl);
 	return 0;
 }
-- 
2.22.0

